features/ubuntu_pro.feature

```diff
@@ -636,14 +629,26 @@ Feature: Command behaviour when auto-attached in an ubuntu PRO image
         log_level: debug
         log_file: /var/log/ubuntu-advantage.log
         """
-        And I reboot the machine
         And I run `pro status --wait` with sudo
         And I run `pro security-status --format json` with sudo
         Then stdout matches regexp:
         """
         "attached": false
         """
-        When I run `cat /var/log/ubuntu-advantage.log` with sudo
+
+        # Prepare for second boot
+        When I run `cloud-init clean --logs` with sudo
+        And I take a snapshot of the machine
+        And I launch a `clone` machine from the snapshot adding this cloud-init user_data:
+        """
+        #cloud-config
+        ubuntu_advantage:
+          features:
+            disable_auto_attach: False
+          enable:
+          - livepatch
+        """
+        And I run `cat /var/log/ubuntu-advantage.log` with sudo
```

Comment:
Here, after booting from the snapshot, `pycloudlib` is not able to ssh into the instance. I am not sure if I am doing something wrong, or it is a bug within the layer on top of `pycloudlib`. If I connect to the instance via the cloud console, I can verify that it behaved as expected (ua nooped and cloud-init drove the auto attach). I will continue investigating what's going on.

<details>
  <summary>Full log</summary>

```text
Loading config defaults from "./tox.ini"
Using defaults:
          color True
  show_snippets True
   show_skipped False
        dry_run False
    show_source True
   show_timings True
 stdout_capture True
 stderr_capture True
    log_capture True
 logging_format %(asctime)s:%(levelname)s:%(name)s:%(message)s
  logging_level debug
  steps_catalog False
        summary True
          junit True
          stage None
       userdata {}
 default_format pretty
   default_tags 
scenario_outline_annotation_schema {name} -- @{row.id} {examples.name}
more_formatters {}
Using default path "./features"
Trying base directory: /home/aciba/canonical/ubuntu-advantage-client/features
--- job suffix: None
Config options:
  image_clean = True
  destroy_instances = True
  ephemeral_instance = False
  snapshot_strategy = False
  cloud_credentials_path = None
  contract_token = None
  contract_token_staging = None
  contract_token_staging_expired = None
  machine_type = gcp.pro
  private_key_file = None
  private_key_name = uaclient-integration
  reuse_image = None
  debs_path = None
  artifact_dir = /tmp/ua
  install_from = InstallationSource.DAILY
  custom_ppa = None
  custom_ppa_keyid = None
  userdata_file = None
  check_version = None
  sbuild_chroot = None
Feature: Full Auto-Attach Endpoint # features/api_full_auto_attach.feature:1

Feature: Creating golden images based on Cloud Ubuntu Pro instances # features/cloud_pro_clone.feature:1

Feature: Pro Upgrade Daemon only runs in environments where necessary # features/daemon.feature:1

Feature: Command behaviour when auto-attached in an ubuntu PRO image # features/ubuntu_pro.feature:1

  @series.lts @uses.config.machine_type.gcp.pro @uses.config.machine_type.aws.pro @uses.config.machine_type.azure.pro
  Scenario Outline: Auto-attach no-op when cloud-init has ubuntu_advantage on userdata and cloud-init auto attaches -- @1.2 ubuntu release  # features/ubuntu_pro.feature:676
    Given a `bionic` machine with ubuntu-advantage-tools installed and cloud-init upgraded to the latest daily version                      # features/steps/machines.py:138 145.961s
    When I run `systemctl mask ubuntu-advantage.service` with sudo                                                                          # features/steps/shell.py:47 0.614s
    And I create the file `/etc/ubuntu-advantage/uaclient.conf` with the following                                                          # features/steps/files.py:33 0.716s
      """
      contract_url: 'https://contracts.canonical.com'
      data_dir: /var/lib/ubuntu-advantage
      log_level: debug
      log_file: /var/log/ubuntu-advantage.log
      """
    And I run `pro status --wait` with sudo                                                                                                 # features/steps/shell.py:47 1.358s
    And I run `pro security-status --format json` with sudo                                                                                 # features/steps/shell.py:47 3.068s
    Then stdout matches regexp                                                                                                              # features/steps/output.py:43 0.000s
      """
      "attached": false
      """
    When I run `cloud-init clean --logs` with sudo                                                                                          # features/steps/shell.py:47 1.101s
    And I take a snapshot of the machine                                                                                                    # features/steps/machines.py:106 97.282s
    And I launch a `clone` machine from the snapshot adding this cloud-init user_data                                                       # features/steps/machines.py:201 98.612s
      """
      #cloud-config
      ubuntu_advantage:
        features:
          disable_auto_attach: False
        enable:
        - livepatch
      """
    And I run `cat /var/log/ubuntu-advantage.log` with sudo                                                                                 # features/steps/shell.py:47 60.293s
      Traceback (most recent call last):
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/pycloudlib/instance.py", line 420, in _ssh_connect
          key_filename=self.key_pair.private_key_path,
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/paramiko/client.py", line 349, in connect
          retry_on_signal(lambda: sock.connect(addr))
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/paramiko/util.py", line 279, in retry_on_signal
          return function()
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/paramiko/client.py", line 349, in <lambda>
          retry_on_signal(lambda: sock.connect(addr))
      socket.timeout: timed out
      
      The above exception was the direct cause of the following exception:
      
      Traceback (most recent call last):
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/behave/model.py", line 1329, in run
          match.run(runner.context)
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/behave/matchers.py", line 98, in run
          self.func(context, *args, **kwargs)
        File "/home/aciba/canonical/ubuntu-advantage-client/features/steps/shell.py", line 63, in when_i_run_command
          result = context.instances[instance_name].execute(full_cmd, stdin=stdin)
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/pycloudlib/instance.py", line 249, in execute
          return self._run_command(command, stdin, **kwargs)
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/pycloudlib/instance.py", line 206, in _run_command
          return self._ssh(list(command), stdin=stdin)
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/pycloudlib/instance.py", line 364, in _ssh
          client = self._ssh_connect()
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/pycloudlib/instance.py", line 430, in _ssh_connect
          raise SSHException from e
      paramiko.ssh_exception.SSHException
      
      Captured stdout:
      HOOK-ERROR in after_step: SSHException: 
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/behave/runner.py", line 545, in run_hook
          self.hooks[name](context, *args)
        File "features/environment.py", line 527, in after_step
          ["cat", log_file], use_sudo=True
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/pycloudlib/instance.py", line 249, in execute
          return self._run_command(command, stdin, **kwargs)
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/pycloudlib/instance.py", line 206, in _run_command
          return self._ssh(list(command), stdin=stdin)
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/pycloudlib/instance.py", line 364, in _ssh
          client = self._ssh_connect()
        File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/pycloudlib/instance.py", line 430, in _ssh_connect
          raise SSHException from e
      
      Captured logging:
      2022-09-30 10:57:52,032:INFO:root:-- user data:
      #cloud-config
      bootcmd:
       - systemctl stop ua-auto-attach.service
      runcmd:
        - "printf \"Package: *\nPin: release o=LP-PPA-ua-client-daily\nPin-Priority: 1001\n\" > /etc/apt/preferences.d/uaclient"
      apt:
        sources:
          ua-tools-ppa:
              source: deb http://ppa.launchpad.net/ua-client/daily/ubuntu $RELEASE main
              keyid: 6E34E7116C0BC933
      
      2022-09-30 10:57:52,032:INFO:root:--- Launching VM to create a base image with ubuntu-advantage
      2022-09-30 10:57:52,032:DEBUG:pycloudlib.cloud.GCE:finding daily Ubuntu image for bionic
      2022-09-30 10:57:52,035:DEBUG:google_auth_httplib2:Making request: POST https://oauth2.googleapis.com/token
      2022-09-30 10:57:52,973:DEBUG:pycloudlib.cloud.GCE:Fetched entire image list matching "ubuntu-pro-1804-bionic-*" in 1 requests
      2022-09-30 10:57:52,973:DEBUG:pycloudlib.cloud.GCE:Found image name: ubuntu-pro-1804-bionic-v20220902
      2022-09-30 10:57:52,973:INFO:root:--- Launching GCP image projects/ubuntu-os-pro-cloud/global/images/3394670333498746152(bionic)
      2022-09-30 10:57:55,229:DEBUG:google.auth._default:Checking /home/aciba/.config/gcloud/application_default_credentials.json for explicit credentials as part of auth process...
      2022-09-30 10:57:55,229:DEBUG:google.auth._default:Explicit credentials path /home/aciba/.config/gcloud/application_default_credentials.json is the same as Cloud SDK credentials path, fall back to Cloud SDK credentials flow...
      2022-09-30 10:57:55,229:DEBUG:google.auth._default:Checking Cloud SDK credentials as part of auth process...
      2022-09-30 10:58:02,310:DEBUG:google_auth_httplib2:Making request: POST https://oauth2.googleapis.com/token
      2022-09-30 10:58:03,539:INFO:pycloudlib.instance:_wait_for_execute to complete
      2022-09-30 10:58:35,800:INFO:paramiko.transport:Connected (version 2.0, client OpenSSH_7.6p1)
      2022-09-30 10:58:36,995:INFO:paramiko.transport:Authentication (publickey) successful!
      2022-09-30 10:58:39,891:INFO:pycloudlib.instance:_wait_for_cloudinit to complete
      2022-09-30 10:58:39,891:INFO:pycloudlib.instance:executing: which systemctl
      2022-09-30 10:58:39,891:DEBUG:pycloudlib.instance:executing: which systemctl
      2022-09-30 10:59:04,160:INFO:pycloudlib.instance:executing: cloud-init status --wait --long
      2022-09-30 10:59:04,160:DEBUG:pycloudlib.instance:waiting for start
      2022-09-30 10:59:05,287:INFO:root:--- gcp instance launched: i0-uaclient-ci-0930-0857-dev-lwm159u9. Waiting for ssh access
      2022-09-30 10:59:21,672:INFO:pycloudlib.instance:_wait_for_execute to complete
      2022-09-30 10:59:22,081:INFO:pycloudlib.instance:_wait_for_cloudinit to complete
      2022-09-30 10:59:22,081:INFO:pycloudlib.instance:executing: which systemctl
      2022-09-30 10:59:22,081:DEBUG:pycloudlib.instance:executing: which systemctl
      2022-09-30 10:59:23,412:INFO:pycloudlib.instance:executing: cloud-init status --wait --long
      2022-09-30 10:59:23,412:DEBUG:pycloudlib.instance:waiting for start
      2022-09-30 10:59:24,539:INFO:root:--- Installing cloud-init PPA: ppa:cloud-init-dev/daily
      2022-09-30 10:59:24,539:INFO:pycloudlib.instance:executing: sh -c 'sudo add-apt-repository ppa:cloud-init-dev/daily -y'
      2022-09-30 10:59:24,539:DEBUG:pycloudlib.instance:executing: sh -c 'sudo add-apt-repository ppa:cloud-init-dev/daily -y'
      2022-09-30 10:59:36,725:INFO:pycloudlib.instance:executing: sh -c 'sudo apt-get update -q'
      2022-09-30 10:59:36,725:DEBUG:pycloudlib.instance:executing: sh -c 'sudo apt-get update -q'
      2022-09-30 10:59:41,640:INFO:pycloudlib.instance:executing: sh -c 'sudo apt-get install -qy cloud-init'
      2022-09-30 10:59:41,640:DEBUG:pycloudlib.instance:executing: sh -c 'sudo apt-get install -qy cloud-init'
      2022-09-30 10:59:49,219:INFO:pycloudlib.instance:executing: sh -c 'sudo cloud-init clean --logs'
      2022-09-30 10:59:49,219:DEBUG:pycloudlib.instance:executing: sh -c 'sudo cloud-init clean --logs'
      2022-09-30 10:59:50,290:DEBUG:google.auth._default:Checking /home/aciba/.config/gcloud/application_default_credentials.json for explicit credentials as part of auth process...
      2022-09-30 10:59:50,290:DEBUG:google.auth._default:Explicit credentials path /home/aciba/.config/gcloud/application_default_credentials.json is the same as Cloud SDK credentials path, fall back to Cloud SDK credentials flow...
      2022-09-30 10:59:50,290:DEBUG:google.auth._default:Checking Cloud SDK credentials as part of auth process...
      2022-09-30 10:59:50,846:INFO:pycloudlib.instance:executing: sh -c 'systemctl is-system-running --wait'
      2022-09-30 10:59:50,846:DEBUG:pycloudlib.instance:executing: sh -c 'systemctl is-system-running --wait'
      2022-09-30 10:59:51,091:DEBUG:google_auth_httplib2:Making request: POST https://oauth2.googleapis.com/token
      2022-09-30 10:59:52,903:INFO:paramiko.transport:Connected (version 2.0, client OpenSSH_7.6p1)
      2022-09-30 10:59:54,030:INFO:paramiko.transport:Authentication (publickey) successful!
      2022-09-30 10:59:54,849:INFO:pycloudlib.instance:executing: sh -c 'sudo apt-get update'
      2022-09-30 10:59:54,850:DEBUG:pycloudlib.instance:executing: sh -c 'sudo apt-get update'
      2022-09-30 10:59:58,126:INFO:pycloudlib.instance:executing: sh -c 'sudo DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-downgrades -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" ubuntu-advantage-tools ubuntu-advantage-pro'
      2022-09-30 10:59:58,127:DEBUG:pycloudlib.instance:executing: sh -c 'sudo DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-downgrades -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" ubuntu-advantage-tools ubuntu-advantage-pro'
      2022-09-30 11:00:05,807:INFO:pycloudlib.instance:executing: sh -c 'sh -c "printf '"'"'features:
        disable_auto_attach: true
      '"'"' > /var/tmp/uaclient.conf"'
      2022-09-30 11:00:05,807:DEBUG:pycloudlib.instance:executing: sh -c 'sh -c "printf '"'"'features:
        disable_auto_attach: true
      '"'"' > /var/tmp/uaclient.conf"'
      2022-09-30 11:00:06,421:INFO:pycloudlib.instance:executing: sh -c 'sudo -- sh -c "cat /var/tmp/uaclient.conf >> /etc/ubuntu-advantage/uaclient.conf"'
      2022-09-30 11:00:06,421:DEBUG:pycloudlib.instance:executing: sh -c 'sudo -- sh -c "cat /var/tmp/uaclient.conf >> /etc/ubuntu-advantage/uaclient.conf"'
      2022-09-30 11:00:07,034:INFO:pycloudlib.instance:executing: sh -c 'sudo pro status --wait'
      2022-09-30 11:00:07,035:DEBUG:pycloudlib.instance:executing: sh -c 'sudo pro status --wait'
      2022-09-30 11:00:09,596:INFO:pycloudlib.instance:executing: sh -c 'sudo pro detach --assume-yes'
      2022-09-30 11:00:09,596:DEBUG:pycloudlib.instance:executing: sh -c 'sudo pro detach --assume-yes'
      2022-09-30 11:00:17,070:INFO:pycloudlib.instance:executing: sh -c 'pro version'
      2022-09-30 11:00:17,070:DEBUG:pycloudlib.instance:executing: sh -c 'pro version'
      2022-09-30 11:00:17,992:INFO:root:--- 28.0-1838~ga0c0499b~ubuntu18.04.1
      2022-09-30 11:00:17,993:INFO:root:--- instance ip: 34.94.103.186
      2022-09-30 11:00:17,993:INFO:pycloudlib.instance:executing: sudo systemctl mask ubuntu-advantage.service
      2022-09-30 11:00:17,993:DEBUG:pycloudlib.instance:executing: sudo systemctl mask ubuntu-advantage.service
      2022-09-30 11:00:18,606:DEBUG:root:stdout: 
      2022-09-30 11:00:18,606:DEBUG:root:stderr: Created symlink /etc/systemd/system/ubuntu-advantage.service → /dev/null.
      2022-09-30 11:00:18,608:INFO:pycloudlib.instance:executing: sudo sh -c 'printf '"'"'contract_url: '"'"'https://contracts.canonical.com'"'"'
      data_dir: /var/lib/ubuntu-advantage
      log_level: debug
      log_file: /var/log/ubuntu-advantage.log'"'"' > /etc/ubuntu-advantage/uaclient.conf'
      2022-09-30 11:00:18,608:DEBUG:pycloudlib.instance:executing: sudo sh -c 'printf '"'"'contract_url: '"'"'https://contracts.canonical.com'"'"'
      data_dir: /var/lib/ubuntu-advantage
      log_level: debug
      log_file: /var/log/ubuntu-advantage.log'"'"' > /etc/ubuntu-advantage/uaclient.conf'
      2022-09-30 11:00:19,323:DEBUG:root:stdout: 
      2022-09-30 11:00:19,323:DEBUG:root:stderr: 
      2022-09-30 11:00:19,324:INFO:pycloudlib.instance:executing: sudo pro status --wait
      2022-09-30 11:00:19,324:DEBUG:pycloudlib.instance:executing: sudo pro status --wait
      2022-09-30 11:00:20,681:DEBUG:root:stdout: SERVICE          AVAILABLE  DESCRIPTION
      cc-eal           yes        Common Criteria EAL2 Provisioning Packages
      cis              yes        Security compliance and audit tools
      esm-apps         yes        Expanded Security Maintenance for Applications
      esm-infra        yes        Expanded Security Maintenance for Infrastructure
      fips             yes        NIST-certified core packages
      fips-updates     yes        NIST-certified core packages with priority security updates
      livepatch        yes        Canonical Livepatch service
      ros              yes        Security Updates for the Robot Operating System
      ros-updates      yes        All Updates for the Robot Operating System
      
      This machine is not attached to an Ubuntu Pro subscription.
      See https://ubuntu.com/pro
      2022-09-30 11:00:20,681:DEBUG:root:stderr: 
      2022-09-30 11:00:20,682:INFO:pycloudlib.instance:executing: sudo pro security-status --format json
      2022-09-30 11:00:20,682:DEBUG:pycloudlib.instance:executing: sudo pro security-status --format json
      2022-09-30 11:00:23,749:DEBUG:root:stdout: {"_schema_version": "0.1", "livepatch": {"fixed_cves": []}, "packages": [{"download_size": 53540, "origin": "security.ubuntu.com", "package": "bind9-host", "service_name": "standard-security", "status": "upgrade_available", "version": "1:9.11.3+dfsg-1ubuntu1.18"}, {"download_size": 145100, "origin": "security.ubuntu.com", "package": "dnsutils", "service_name": "standard-security", "status": "upgrade_available", "version": "1:9.11.3+dfsg-1ubuntu1.18"}, {"download_size": 27620, "origin": "security.ubuntu.com", "package": "libbind9-160", "service_name": "standard-security", "status": "upgrade_available", "version": "1:9.11.3+dfsg-1ubuntu1.18"}, {"download_size": 748752, "origin": "security.ubuntu.com", "package": "libdns-export1100", "service_name": "standard-security", "status": "upgrade_available", "version": "1:9.11.3+dfsg-1ubuntu1.18"}, {"download_size": 958936, "origin": "security.ubuntu.com", "package": "libdns1100", "service_name": "standard-security", "status": "upgrade_available", "version": "1:9.11.3+dfsg-1ubuntu1.18"}, {"download_size": 19096, "origin": "security.ubuntu.com", "package": "libirs160", "service_name": "standard-security", "status": "upgrade_available", "version": "1:9.11.3+dfsg-1ubuntu1.18"}, {"download_size": 163212, "origin": "security.ubuntu.com", "package": "libisc-export169", "service_name": "standard-security", "status": "upgrade_available", "version": "1:9.11.3+dfsg-1ubuntu1.18"}, {"download_size": 236888, "origin": "security.ubuntu.com", "package": "libisc169", "service_name": "standard-security", "status": "upgrade_available", "version": "1:9.11.3+dfsg-1ubuntu1.18"}, {"download_size": 17880, "origin": "security.ubuntu.com", "package": "libisccc160", "service_name": "standard-security", "status": "upgrade_available", "version": "1:9.11.3+dfsg-1ubuntu1.18"}, {"download_size": 48344, "origin": "security.ubuntu.com", "package": "libisccfg160", "service_name": "standard-security", "status": "upgrade_available", "version": "1:9.11.3+dfsg-1ubuntu1.18"}, {"download_size": 34528, "origin": "security.ubuntu.com", "package": "liblwres160", "service_name": "standard-security", "status": "upgrade_available", "version": "1:9.11.3+dfsg-1ubuntu1.18"}, {"download_size": 104984, "origin": "security.ubuntu.com", "package": "libnss-systemd", "service_name": "standard-security", "status": "upgrade_available", "version": "237-3ubuntu10.56"}, {"download_size": 107352, "origin": "security.ubuntu.com", "package": "libpam-systemd", "service_name": "standard-security", "status": "upgrade_available", "version": "237-3ubuntu10.56"}, {"download_size": 498632, "origin": "security.ubuntu.com", "package": "libsqlite3-0", "service_name": "standard-security", "status": "upgrade_available", "version": "3.22.0-1ubuntu0.6"}, {"download_size": 204704, "origin": "security.ubuntu.com", "package": "libsystemd0", "service_name": "standard-security", "status": "upgrade_available", "version": "237-3ubuntu10.56"}, {"download_size": 53992, "origin": "security.ubuntu.com", "package": "libudev1", "service_name": "standard-security", "status": "upgrade_available", "version": "237-3ubuntu10.56"}, {"download_size": 293656, "origin": "security.ubuntu.com", "package": "sosreport", "service_name": "standard-security", "status": "upgrade_available", "version": "4.3-1ubuntu0.18.04.2"}, {"download_size": 2915672, "origin": "security.ubuntu.com", "package": "systemd", "service_name": "standard-security", "status": "upgrade_available", "version": "237-3ubuntu10.56"}, {"download_size": 11784, "origin": "security.ubuntu.com", "package": "systemd-sysv", "service_name": "standard-security", "status": "upgrade_available", "version": "237-3ubuntu10.56"}, {"download_size": 1100316, "origin": "security.ubuntu.com", "package": "udev", "service_name": "standard-security", "status": "upgrade_available", "version": "237-3ubuntu10.56"}, {"download_size": 1153700, "origin": "security.ubuntu.com", "package": "vim", "service_name": "standard-security", "status": "upgrade_available", "version": "2:8.0.1453-1ubuntu1.9"}, {"download_size": 71336, "origin": "security.ubuntu.com", "package": "vim-common", "service_name": "standard-security", "status": "upgrade_available", "version": "2:8.0.1453-1ubuntu1.9"}, {"download_size": 5435976, "origin": "security.ubuntu.com", "package": "vim-runtime", "service_name": "standard-security", "status": "upgrade_available", "version": "2:8.0.1453-1ubuntu1.9"}, {"download_size": 476900, "origin": "security.ubuntu.com", "package": "vim-tiny", "service_name": "standard-security", "status": "upgrade_available", "version": "2:8.0.1453-1ubuntu1.9"}, {"download_size": 50168, "origin": "security.ubuntu.com", "package": "xxd", "service_name": "standard-security", "status": "upgrade_available", "version": "2:8.0.1453-1ubuntu1.9"}, {"download_size": 95220, "origin": "esm.ubuntu.com", "package": "krb5-locales", "service_name": "esm-apps", "status": "pending_attach", "version": "1.16-2ubuntu0.2+esm1"}, {"download_size": 157820, "origin": "esm.ubuntu.com", "package": "libgssapi-krb5-2", "service_name": "esm-apps", "status": "pending_attach", "version": "1.16-2ubuntu0.2+esm1"}, {"download_size": 121152, "origin": "esm.ubuntu.com", "package": "libk5crypto3", "service_name": "esm-apps", "status": "pending_attach", "version": "1.16-2ubuntu0.2+esm1"}, {"download_size": 314548, "origin": "esm.ubuntu.com", "package": "libkrb5-3", "service_name": "esm-apps", "status": "pending_attach", "version": "1.16-2ubuntu0.2+esm1"}, {"download_size": 62940, "origin": "esm.ubuntu.com", "package": "libkrb5support0", "service_name": "esm-apps", "status": "pending_attach", "version": "1.16-2ubuntu0.2+esm1"}], "summary": {"num_esm_apps_packages": 0, "num_esm_apps_updates": 5, "num_esm_infra_packages": 0, "num_esm_infra_updates": 0, "num_installed_packages": 525, "num_main_packages": 517, "num_multiverse_packages": 0, "num_restricted_packages": 0, "num_standard_security_updates": 25, "num_third_party_packages": 3, "num_universe_packages": 5, "num_unknown_packages": 0, "reboot_required": "no", "ua": {"attached": false, "enabled_services": [], "entitled_services": []}}}
      2022-09-30 11:00:23,749:DEBUG:root:stderr: 
      2022-09-30 11:00:23,751:INFO:pycloudlib.instance:executing: sudo cloud-init clean --logs
      2022-09-30 11:00:23,751:DEBUG:pycloudlib.instance:executing: sudo cloud-init clean --logs
      2022-09-30 11:00:24,851:DEBUG:root:stdout: 
      2022-09-30 11:00:24,851:DEBUG:root:stderr: 
      2022-09-30 11:02:02,135:DEBUG:root:--- Snapshot created: projects/canonical-348509/global/images/i0-uaclient-ci-0930-0857-dev-lwm159u9-image
      2022-09-30 11:02:02,135:INFO:root:--- Launching GCP image projects/canonical-348509/global/images/i0-uaclient-ci-0930-0857-dev-lwm159u9-image(bionic)
      2022-09-30 11:02:04,797:DEBUG:google.auth._default:Checking /home/aciba/.config/gcloud/application_default_credentials.json for explicit credentials as part of auth process...
      2022-09-30 11:02:04,797:DEBUG:google.auth._default:Explicit credentials path /home/aciba/.config/gcloud/application_default_credentials.json is the same as Cloud SDK credentials path, fall back to Cloud SDK credentials flow...
      2022-09-30 11:02:04,797:DEBUG:google.auth._default:Checking Cloud SDK credentials as part of auth process...
      2022-09-30 11:02:21,591:DEBUG:google_auth_httplib2:Making request: POST https://oauth2.googleapis.com/token
      2022-09-30 11:02:23,844:INFO:pycloudlib.instance:_wait_for_execute to complete
      2022-09-30 11:02:55,894:INFO:paramiko.transport:Connected (version 2.0, client OpenSSH_7.6p1)
      2022-09-30 11:02:57,124:INFO:paramiko.transport:Authentication (publickey) successful!
      2022-09-30 11:03:00,456:INFO:pycloudlib.instance:_wait_for_cloudinit to complete
      2022-09-30 11:03:00,456:INFO:pycloudlib.instance:executing: which systemctl
      2022-09-30 11:03:00,456:DEBUG:pycloudlib.instance:executing: which systemctl
      2022-09-30 11:03:18,918:INFO:pycloudlib.instance:executing: cloud-init status --wait --long
      2022-09-30 11:03:18,918:DEBUG:pycloudlib.instance:waiting for start
      2022-09-30 11:03:20,061:INFO:root:--- gcp instance launched: i1-uaclient-ci-0930-0857-dev-lwm159u9. Waiting for ssh access
      2022-09-30 11:03:36,753:INFO:pycloudlib.instance:_wait_for_execute to complete
      2022-09-30 11:03:37,160:INFO:pycloudlib.instance:_wait_for_cloudinit to complete
      2022-09-30 11:03:37,160:INFO:pycloudlib.instance:executing: which systemctl
      2022-09-30 11:03:37,160:DEBUG:pycloudlib.instance:executing: which systemctl
      2022-09-30 11:03:38,379:INFO:pycloudlib.instance:executing: cloud-init status --wait --long
      2022-09-30 11:03:38,379:DEBUG:pycloudlib.instance:waiting for start
      2022-09-30 11:03:39,620:INFO:pycloudlib.instance:executing: cloud-init status --wait --long
      2022-09-30 11:03:39,620:DEBUG:pycloudlib.instance:executing: cloud-init status --wait --long
      2022-09-30 11:03:40,747:INFO:root:--- cloud-init succeeded
      2022-09-30 11:03:40,747:INFO:root:--- Instance launched rom snapshot: instance_id=651613251416297060, instance_name=ubuntu-behave-test-1664528522135802-clone
      2022-09-30 11:03:40,748:INFO:pycloudlib.instance:executing: sudo cat /var/log/ubuntu-advantage.log
      2022-09-30 11:03:40,748:DEBUG:pycloudlib.instance:executing: sudo cat /var/log/ubuntu-advantage.log
      2022-09-30 11:04:41,041:INFO:root:-- pull instance:/etc/ubuntu-advantage/uaclient.log /tmp/ua/ubuntu_pro.feature_651/uaclient.log
      2022-09-30 11:04:41,041:INFO:pycloudlib.instance:executing: sudo -- cat /etc/ubuntu-advantage/uaclient.log
      2022-09-30 11:04:41,041:DEBUG:pycloudlib.instance:executing: sudo -- cat /etc/ubuntu-advantage/uaclient.log

CLEANUP-ERROR in _cleanup_instance: SSLError: [SSL: KRB5_S_TKT_NYV] unexpected eof while reading (_ssl.c:2570)
Traceback (most recent call last):
  File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/behave/runner.py", line 222, in _do_cleanups
    cleanup_func()
  File "/home/aciba/canonical/ubuntu-advantage-client/features/util.py", line 303, in _cleanup_instance
    context.instances[instance_name].delete(wait=False)
  File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/pycloudlib/gce/instance.py", line 99, in delete
    project=self.project, zone=self.zone, instance=self.instance_id
  File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/googleapiclient/_helpers.py", line 130, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/googleapiclient/http.py", line 932, in execute
    headers=self.headers,
  File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/googleapiclient/http.py", line 222, in _retry_request
    raise exception
  File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/googleapiclient/http.py", line 191, in _retry_request
    resp, content = http.request(uri, method, *args, **kwargs)
  File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/google_auth_httplib2.py", line 225, in request
    **kwargs
  File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/httplib2/__init__.py", line 1702, in request
    conn, authority, uri, request_uri, method, body, headers, redirections, cachekey,
  File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/httplib2/__init__.py", line 1421, in _request
    (response, content) = self._conn_request(conn, request_uri, method, body, headers)
  File "/home/aciba/canonical/ubuntu-advantage-client/.tox/behave-gcppro-18.04/lib/python3.7/site-packages/httplib2/__init__.py", line 1373, in _conn_request
    response = conn.getresponse()
  File "/home/aciba/.pyenv/versions/3.7.13/lib/python3.7/http/client.py", line 1373, in getresponse
    response.begin()
  File "/home/aciba/.pyenv/versions/3.7.13/lib/python3.7/http/client.py", line 319, in begin
    version, status, reason = self._read_status()
  File "/home/aciba/.pyenv/versions/3.7.13/lib/python3.7/http/client.py", line 280, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/home/aciba/.pyenv/versions/3.7.13/lib/python3.7/socket.py", line 589, in readinto
    return self._sock.recv_into(b)
  File "/home/aciba/.pyenv/versions/3.7.13/lib/python3.7/ssl.py", line 1071, in recv_into
    return self.read(nbytes, buffer)
  File "/home/aciba/.pyenv/versions/3.7.13/lib/python3.7/ssl.py", line 929, in read
    return self._sslobj.read(len, buffer)
ssl.SSLError: [SSL: KRB5_S_TKT_NYV] unexpected eof while reading (_ssl.c:2570)
SKIP Scenario Auto-attach no-op when cloud-init has ubuntu_advantage on userdata and cloud-init auto attaches -- @1.3 ubuntu release: Skipping scenario outline series `focal`. Cmdline provided @series tags: {'bionic', 'lts'}
SKIP Scenario Auto-attach no-op when cloud-init has ubuntu_advantage on userdata and cloud-init auto attaches -- @1.4 ubuntu release: Skipping scenario outline series `jammy`. Cmdline provided @series tags: {'bionic', 'lts'}
    Then stdout matches regexp                                                                                                              # None
      """
      cloud-init userdata has ubuntu-advantage key.
      """
    And stdout matches regexp                                                                                                               # None
      """
      Skipping auto-attach and deferring to cloud-init to setup and configure auto-attach
      """
    When I run `cloud-init status` with sudo                                                                                                # None
    Then stdout matches regexp                                                                                                              # None
      """
      status: done
      """
    When I run `pro status --wait` with sudo                                                                                                # None
    And I run `pro security-status --format json` with sudo                                                                                 # None
    Then stdout matches regexp                                                                                                              # None
      """
      "attached": true
      """

```  

</details>

Reply:
Never mind, I was executing commands on the incorrect instance. Fixed.